v4/newrelic/segments.go

```diff
@@ -448,13 +456,22 @@ func (s *MessageProducerSegment) End() {
 		return
 	}
 	if s.StartTime.isEnded() {
+		logAlreadyEnded(s.StartTime, s.name())
 		return
 	}
 	s.addRequiredAttributes(s.StartTime.Span.SetAttribute)
 	s.StartTime.Span.SetName(s.name())
 	s.StartTime.end()
 }
 
+func logAlreadyEnded(st SegmentStartTime, name string) {
+	if st.span != nil {
```

Comment:
the path to the logger from here is `st.span.thread.logger`, but calling `st.thread.logDebug` handles the check for the nil receiver (where the receiver is the thread) so therefore here I am only checking the `st.span` for nil.

Reply:
If the path to the logger is `st.span.thread.logger`, why isn't the call `st.span.thread.logDebug`?

Reply:
In this case the `span` is embedded in the `SegmentStartTime` so you can call `st.thread` directly, however if the `span` is nil it will still panic (even though it's not explicitly in the call).